src/baldr/complexrestriction.cc

```diff
@@ -0,0 +1,40 @@
+#include "baldr/complexrestriction.h"
+
+namespace valhalla {
+namespace baldr {
+
+bool CheckPatchPathForRestrictions(
```

Comment:
i'm wondering if you can just delete this function all together and use `std::search` which will return an interator in the super vector to the start of the subvector if it was found and super vector.end otherwise. basically the body of the function could just be this:

```c++
for(const auto& restriction : restrictions)
  if(std::search(patch_path.cbegin(), patch_path.cend(), restriction.begin(), restriction.end()) != patch_path.cend())
    return true;
return false;
```

Reply:
Interesting, this would be much preferable to a hand-cranked algorithm. I'll try this out.

Reply:
This worked fine!